spec/models/user_spec.rb

```diff
@@ -0,0 +1,160 @@
+require 'rails_helper'
+require 'pry'
+
+describe User do
+
+  let(:user){ build(:user) }
+
+  it "is valid with default attributes" do
+    expect(user).to be_valid
+  end
+
+  it "is not valid without a first_name" do
+    user.first_name = nil
+    expect(user).to be_invalid
+  end
+
+  it "is not valid without a last_name" do
+    user.last_name = nil
+    expect(user).to be_invalid
+  end
+
+  it "is not valid without a password" do
+    user.password = nil
+    expect(user).to be_invalid
+  end
+
+  it 'is not valid if password is eight spaces' do
+    user.password = "        "
+    expect(user).to be_invalid
+  end
+
+  # This should not pass either
```

Comment:
change the values of both password and password_confirmation (as user has not yet been saved) to create a new password after save.